lib/google_sign_in_desktop.dart

```diff
@@ -456,6 +460,91 @@ class GoogleSignInDesktop extends GoogleSignInPlatform {
       await server.close();
     }
   }
+
+  @override
+  // ignore: avoid_renaming_method_parameters
+  Future<bool> requestScopes(List<String> requestedScopes) async {
+    if (requestedScopes.isEmpty) {
+      return false;
+    }
+
+    // Keep track of the previous scopes in case we need to revert
+    final previousScopes = _scopes;
+
+    try {
+      // Merge the current scopes with the requested scopes.
+      _scopes =
+          _buildEffectiveScopes({..._scopes, ...requestedScopes}.toList());
+
+      GoogleSignInDesktopTokenData tokenData;
+      final String? email = _userData?.email;
+      if (email == null) {
+        // Not signed in, so fresh sign in with the new scopes
+        final signInRes = await _signIn();
+        tokenData = signInRes.$1;
+      } else {
+        // Already signed in, so get the token data to see if the scopes are already granted
+        tokenData = await getTokens(email: email, shouldRecoverAuth: false);
+      }
```

Comment:
Could we use the `_tokenDataStore` similar to what has been done elsewhere?

```suggestion
      var tokenData = await _tokenDataStore.get();
      if (tokenData == null) {
        tokenData = (await _signIn()).$1;

        return _hasGrantedAllScopes(
          queryScopes: requestedScopes,
          grantedScopes: tokenData.scopes ?? [],
        );
      }
```